### PR TITLE
Bug fix: expr_visitor was not visiting aggregate filter expressions

### PR DESCRIPTION
**Pull request by andygrove**
# Which issue does this PR close?

Closes https://github.com/apache/arrow-datafusion/issues/3546

 # Rationale for this change

# What changes are included in this PR?

# Are there any user-facing changes?

### CODE REVIEW

File: datafusion/optimizer/src/projection_push_down.rs [hunk 1]
```diff
@@ -990,6 +990,32 @@ mod tests {
...
+
+        let plan = LogicalPlanBuilder::from(table_scan)
+            .aggregate(
+                vec![col("a")],
+                vec![count(col("b")), aggr_with_filter.alias("count2")],
+            )?
+            .build()?;
+
+        let expected = "Aggregate: groupBy=[[#test.a]], aggr=[[COUNT(#test.b), COUNT(#test.b) FILTER (WHERE #c > Int32(42)) AS count2]]\
+        \n  TableScan: test projection=[a, b, c]";
```
Comment (andygrove) [inline datafusion/optimizer/src/projection_push_down.rs:1012]: Prior to the fix, the table scan projection was `[a, b]`

**Comment by andygrove** [comment]
@alamb @tustvold please take a look when you can - this came up through Dask SQL testing

**Review by Dandandan — APPROVED** [review]

**Comment by andygrove** [comment]
Thanks for the review @Dandandan! I am going to go ahead and merge since this seems uncontroversial and it unblocks a Dask SQL release

**Review by alamb — APPROVED** [review]

**Comment by alamb** [comment]
Thanks @andygrove

**Pull request merged**